delete.php

```diff
@@ -25,5 +25,16 @@
 
 require_sesskey();
 
-$DB->delete_records('cpdlogbook_entries', ['id' => $id]);
+// If the cpdlogbook doesn't exist.
+if (! $cpdlogbook = $DB->get_record('cpdlogbook', ['id' => $cm->instance])) {
```

Comment:
Can you simply remove the cmid param completely and swap the order of the lookups so you explicitely grab the cmid from the logbook entry?

Reply:
Would this be done using the `get_coursemodule_from_instance` function?

Reply:
Sounds about right, only one way to find out :)